app/uk/gov/hmrc/thirdpartyorchestrator/services/ApplicationFetcher.scala

```diff
@@ -23,27 +23,37 @@ import scala.util.control.NonFatal
 import uk.gov.hmrc.http.HeaderCarrier
 
 import uk.gov.hmrc.apiplatform.modules.applications.core.domain.models.ApplicationResponse
-import uk.gov.hmrc.apiplatform.modules.common.domain.models.ApplicationId
+import uk.gov.hmrc.apiplatform.modules.common.domain.models.{ApplicationId, ClientId}
 import uk.gov.hmrc.thirdpartyorchestrator.connectors.EnvironmentAwareThirdPartyApplicationConnector
 import uk.gov.hmrc.thirdpartyorchestrator.utils.ApplicationLogger
 
 @Singleton
-class ApplicationByIdFetcher @Inject() (
+class ApplicationFetcher @Inject() (
     thirdPartyApplicationConnector: EnvironmentAwareThirdPartyApplicationConnector
   )(implicit ec: ExecutionContext
   ) extends ApplicationLogger {
 
-  def fetchApplication(id: ApplicationId)(implicit hc: HeaderCarrier): Future[Option[ApplicationResponse]] = {
-    val subordinateApp: Future[Option[ApplicationResponse]] = thirdPartyApplicationConnector.subordinate.fetchApplication(id) recover recoverWithDefault(None)
-    val principalApp: Future[Option[ApplicationResponse]]   = thirdPartyApplicationConnector.principal.fetchApplication(id)
+  def fetchApplication(applicationId: ApplicationId)(implicit hc: HeaderCarrier): Future[Option[ApplicationResponse]] = {
+    val subordinateApp: Future[Option[ApplicationResponse]] = thirdPartyApplicationConnector.subordinate.fetchApplication(applicationId) recover recoverWithDefault(None)
+    val principalApp: Future[Option[ApplicationResponse]]   = thirdPartyApplicationConnector.principal.fetchApplication(applicationId)
 
     for {
       subordinate <- subordinateApp
       principal   <- principalApp
     } yield principal.orElse(subordinate)
   }
 
-  def recoverWithDefault[T](default: T): PartialFunction[Throwable, T] = {
+  def fetchApplicationByClientId(clientId: ClientId)(implicit hc: HeaderCarrier): Future[Option[ApplicationResponse]] = {
```

Comment:
The other method is just called `fetchApplication` whereas this one is called `fetchApplicationByClientId`
Something about this feel inconsistent.
Maybe they both `fetchApplication` just with differing params or maybe they both be very specific and include the param in the method name, like this new one?

Reply:
Fixed